go/pkg/cas/upload.go

```diff
@@ -0,0 +1,42 @@
+package cas
+
+import "context"
+
+// UploadInput is one of inputs to Client.Upload function.
+//
+// It can be either a reference to a file/dir (see Path) or it can be an
+// in-memory blob (see Content).
+type UploadInput struct {
+	// Path to the file or a directory to upload.
+	// If empty, the Content is uploaded instead.
+	Path string
```

Comment:
Can we set expectations around what type of path this will be? abs or relative to cwd?

Reply:
Done